lang/cs/Org.Apache.REEF.Wake/Remote/Impl/WritableRemoteManagerFactory.cs

```diff
@@ -31,16 +31,18 @@ namespace Org.Apache.REEF.Wake.Impl
     [Obsolete("Need to remove Iwritable and use IstreamingCodec. Please see Jira REEF-295 ", false)]
     public sealed class WritableRemoteManagerFactory
     {
+        private readonly ITcpPortProvider _tcpPortProvider;
         [Inject]
-        private WritableRemoteManagerFactory()
+        private WritableRemoteManagerFactory(ITcpPortProvider tcpPortProvider)
         {
+            _tcpPortProvider = tcpPortProvider;
         }
 
         public IRemoteManager<T> GetInstance<T>(IPAddress localAddress, int port) where T : IWritable
```

Comment:
Here, I really think we should have two methods: One which takes the `port`, the other which doesn't take one. If a port is allowed to be passed, we should adhere to it. We should of course try not to need to pass the port.